dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SqlViewController.java

```diff
@@ -212,25 +214,21 @@ public void getViewPdf( @PathVariable( "uid" ) String uid,
 
     @PostMapping( "/{uid}/execute" )
     @ResponseBody
-    public WebMessage executeView( @PathVariable( "uid" ) String uid,
-        @RequestParam( required = false ) Set<String> var )
+    public WebMessage executeView( @PathVariable( "uid" ) String uid )
+        throws NotFoundException,
+        ConflictException
     {
-        SqlView sqlView = sqlViewService.getSqlViewByUid( uid );
-
-        if ( sqlView == null )
-        {
-            return notFound( "SQL view does not exist: " + uid );
-        }
+        SqlView sqlView = getExistingSQLView( uid );
 
         if ( sqlView.isQuery() )
         {
-            return conflict( "SQL view is a query, no view to create" );
+            throw new ConflictException( "SQL view is a query, no view to create" );
```

Comment:
Do we want to add an `ErrorCode` for this for easier programmatic handling by clients?